pkg/reconciler/domainmapping/resources/ingress.go

```diff
@@ -25,22 +25,16 @@ import (
 	"knative.dev/networking/pkg/apis/networking"
 	netv1alpha1 "knative.dev/networking/pkg/apis/networking/v1alpha1"
 	"knative.dev/pkg/kmeta"
-	pkgnetwork "knative.dev/pkg/network"
 	servingv1alpha1 "knative.dev/serving/pkg/apis/serving/v1alpha1"
 )
 
 // MakeIngress creates an Ingress object for a DomainMapping.
-func MakeIngress(dm *servingv1alpha1.DomainMapping, ingressClass string) *netv1alpha1.Ingress {
+func MakeIngress(dm *servingv1alpha1.DomainMapping, hostName, ingressClass string) *netv1alpha1.Ingress {
 	var (
 		targetServiceName      = dm.Spec.Ref.Name
 		targetServiceNamespace = dm.Spec.Ref.Namespace
 	)
```

Comment:
I think in the short term we should change this to parse the URL and error out if it doesn't end with `network.GetClusterDomainSuffix()` (or w/e we called that).

In the longer term, I think we may need to create an ExternalName service when that doesn't match, but it will be hard to test...

Reply:
yeah, I _think_ for this PR we still require ksvc so that's ok, when we relax the validation we need to worry about stuff that doesn't already have a matching service etc

Reply:
.. either way, added some validation for now